src/Discovery/RetrieveDiscoveryData.cs

```diff
@@ -243,6 +259,44 @@ private List<DiscoveryData> ParsePhysicalDiscoveryData(UserInput userInputObj, M
 
             return data;
         }
+
+        private List<DiscoveryData> ParseImportDiscoveryData(UserInput userInputObj, MachinesImportJSON jsonObj)
+        {
+            List<DiscoveryData> data = new List<DiscoveryData>();
+
+            foreach (var value in jsonObj.Values)
+            {
+                DiscoveryData discoveryDataObj = new DiscoveryData();
+
+                // Assign values
+                discoveryDataObj.MachineName = value.Properties.DisplayName;
+                discoveryDataObj.EnvironmentType = ""; // only 'Dev' is treated as 'Dev' environment
+
+                discoveryDataObj.SoftwareInventory = 0;
+                discoveryDataObj.SqlDiscoveryServerCount = 0;
+                discoveryDataObj.IsSqlServicePresent = false;
+                discoveryDataObj.WebAppCount = 0;
+
+                discoveryDataObj.OperatingSystem = value.Properties.OperatingSystemDetails?.OSName;
+                discoveryDataObj.Cores = value.Properties.NumberOfProcessorCore ?? 0;
+                discoveryDataObj.MemoryInMB = value.Properties.AllocatedMemoryInMB ?? 0.0;
+                discoveryDataObj.TotalDisks = value.Properties.Disks?.Count?? 0;                
+                discoveryDataObj.IpAddress = value.Properties.IPAddresses != null ? string.Join(",", value.Properties.IPAddresses.Select(ip => $"[{ip}]")) + ";" : "";
+                discoveryDataObj.MacAddress = value.Properties.MacAddress;

```

Comment:
Need to verify this, is it given that there can be only one Mac Address?

Reply:
verified. It is a string
![image](https://github.com/user-attachments/assets/811fe96e-e740-4052-b968-9e5444d244ec)